.github/workflows/ci.yml

```diff
@@ -424,59 +421,39 @@ jobs:
   firmware-lints:
     name: Firmware Lints
     runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: git-nix-shell {0}
+    needs: [build, lint]
```

Comment:
the `build` task is for building Haskell stuff only, right? This task doesn't depend on that semantically, firmware linting can potentially also be much faster so it might make sense to remove this dependency to make CI checks faster when working on firmware stuff?